remote-config/src/main/kotlin/com/rakuten/tech/mobile/remoteconfig/ConfigFetcher.kt

```diff
@@ -0,0 +1,59 @@
+package com.rakuten.tech.mobile.remoteconfig
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import java.io.IOException
+import java.lang.IllegalArgumentException
+
+internal class ConfigFetcher(
+    baseUrl: String,
+    appId: String,
+    private val subscriptionKey: String
+) {
+
+    private val client = OkHttpClient()
+    private val requestUrl = try {
+        HttpUrl.get(baseUrl)
+            .newBuilder()
+            .addPathSegments("app/$appId/config")
+            .build()
+    } catch (exception: IllegalArgumentException) {
+        throw InvalidRemoteConfigBaseUrlException(exception)
+    }
+
+    fun fetch(): Map<String, String> {
+        val response = client.newCall(buildFetchRequest())
+            .execute()
```

Comment:
This looks sync but is it async under-the-hood?

Reply:
Nope, `execute` is a blocking call, so this function has to be called asynchronously. In the `ConfigCache` it's being called in a coroutine with `GlobalScope.launch`.

Reply:
Thanks - I now understand that `GlobalScope.launch` is running a coroutine.
Is there a convention for documenting/tagging functions as blocking? I guess it's internal though so it's not really required.

Reply:
Hmm, no I don't think there is really. There's the `suspend` keyword, but it's intended to be used specifically with functions that can suspend a Coroutine. All it does is prevent you from calling the function outside of a Coroutine, so I don't think it's the right use case for here (and my IDE tells me to remove it when I added it).